spec/models/page_spec.rb

```diff
@@ -256,5 +263,62 @@
       expect(Page.new.action_count).to eq 0
     end
   end
+
+  describe 'friendly_id' do
+    let!(:page) { create(:page, title: 'simple slug') }
+
+    it 'generates slug' do
+      expect(page.friendly_id).to eq('simple-slug')
+      expect(page.slug).to        eq('simple-slug')
+    end
+
+    context 'finder' do
+      it 'finds by slug' do
+        expect(Page.find('simple-slug')).to eq page
+      end
+
+      it 'finds by id' do
+        expect(Page.find(page.id)).to eq page
+      end
+
+      it 'finds using friendly.find' do
+        expect(Page.friendly.find('simple-slug')).to eq page
+        expect(Page.friendly.find(page.id)).to eq page
+      end
+    end
+
+    context 'updating title' do
+      before do
+        page.update title: 'Complex Slug', slug: nil
+      end
+
+      it 'updates slug' do
+        expect(page.title).to              eq("Complex Slug")
+        expect(page.reload.friendly_id).to eq('complex-slug')
+      end
+    end
+
+    context 'historical' do
```

Comment:
I definitely think we should use the `historical` module - http://norman.github.io/friendly_id/file.Guide.html#History__Avoiding_404_s_When_Slugs_Change